src/controllers/AuthController.ts

```diff
@@ -106,264 +104,122 @@ class AuthController {
-		// 		success: true,
-		// 		message: `OTP sent successfully`
-		// 	})
-		// } catch (error) {
-		// 	res.status(400).json({
-		// 		status: 400,
-		// 		message: error?.toString(),
-		// 		code: "unexpected_error"
-		// 	})
-		// 	return
-		// }
+		try {
+			const response = new ApiResponse(res)
+			const {email}: SendOtpPayload = req.body
+			//check if user exist
+			const userExists: UserDetails | null = await User.findOne({
+				email,
+				isDeleted: false,
+				isActive: true
+			})
+			if (!userExists) {
+				return response.errorResponse({
+					...errorData.NOT_FOUND,
+					message: "User not found"
+				})
+			}
+
+			// send otp to email
+			await sendOtpToEmail(
+				authCredential.userName,
+				otp,
+				userDetails.firstName
+			)
+
+			return response.successResponse({
+				message: `OTP sent successfully`
+			})
+		} catch (error) {
+			next(error)
+		}
 	}
 
 	public async verifyOtp(req: Request, res: Response, next: NextFunction) {
-		// try {
-		// 	const {hash, otp}: {hash: string; otp: number} = req.body
-		// 	const decryptData: DecryptData = await decryptBycrypto(hash)
-		// 	const {email, userId}: DecryptData = decryptData
-		// 	// check if otp is valid
-		// 	const [verificationData]: VerificationTableData[] =
-		// 		await this.verificationCommonModel.list({
-		// 			value: email,
-		// 			isVerified: false,
-		// 			verificationType: LogInWith.EMAIL,
-		// 			verificationFor: VerificationFor.AUTH,
-		// 			status: true
-		// 		})
-		// 	if (!verificationData) {
-		// 		throw new UnauthorizedException("User not found")
-		// 	}
-		// 	let decoded: any = null
-		// 	try {
-		// 		decoded = jwt.verify(
-		// 			(verificationData?.otp).toString(),
-		// 			process.env.JWT_SECRET_KEY as string
-		// 		)
-		// 	} catch (error) {
-		// 		throw new UnauthorizedException(
-		// 			"Invalid OTP. Please resend and try again."
-		// 		)
-		// 	}
-		// 	if (parseInt(decoded?.otp) !== parseInt(otp.toString())) {
-		// 		throw new UnauthorizedException(
-		// 			"Invalid OTP. Please resend and try again."
-		// 		)
-		// 	}
-		// 	if (
-		// 		new Date(
-		// 			new Date(verificationData.createdAt.toString()).getTime() +
-		// 				parseInt(
-		// 					process.env.OTP_EXPIRATION_IN_MINUTES as string
-		// 				) *
-		// 					60000
-		// 		).getTime() < new Date().getTime()
-		// 	) {
-		// 		throw new UnauthorizedException(
-		// 			"OTP expired. Please resend and try again."
-		// 		)
-		// 	}
-		// 	// mark OTP as used
-		// 	await this.verificationCommonModel.update(
-		// 		{isVerified: true},
-		// 		verificationData.verificationId
-		// 	)
-		// 	return res.json({
-		// 		success: true,
-		// 		message: `OTP verified successfully`
-		// 	})
-		// } catch (error) {
-		// 	res.status(400).json({
-		// 		status: 400,
-		// 		message: error?.toString(),
-		// 		code: "unexpected_error"
-		// 	})
-		// 	return
-		// }
+		try {
+			const response = new ApiResponse(res)
+			const {hash, otp}: {hash: string; otp: number} = req.body
```

Comment:
we will receive email and otp as input from body.
Please use email instead of hash.

Reply:
done